File: Stack/LinkedStack/LinkedStack.java
import java.lang.*;
import java.util.*;


public class LinkedStack<T>
{
    private Node topNode;

    public LinkedStack(){
        topNode = null;
    }
    public LinkedStack(Node stackInput){
        topNode = stackInput;
    }
    //Method runs in O(1)
    public void push(T newData){
        Node newNode = new Node(newData, topNode);
        topNode = newNode;

    }
    //Method runs in O(1)
    public T pop(){
        if (topNode != null){
            T tempData = topNode.getData();
            topNode = topNode.getNext();
            return tempData;
        }
        else
            throw new EmptyStackException();
    }
    //Method runs in O(1)
    public T peek(){
        if (isEmpty())
            throw new EmptyStackException();
        else
            return topNode.getData();
    }
    //Method runs in O(1)
    public boolean isEmpty(){
        return topNode == null;
    }
    //Method runs in O(1)
    public void clear(){
        topNode = null; 
    }
    //Print method runs in O(n)
    public void printStack(){
        Node tempNode = topNode;
        while (tempNode != null)
        {
            System.out.println(tempNode.getData());
            tempNode = tempNode.getNext();
        }
    }
    
    //Node class

    private class Node{
        private T data;
        private Node next;


        private Node(T data){
            this.data = data;
            this.next = null;
        }
        private Node(T data, Node next)
        {
            this.data = data;
            this.next = next;
        }
        private T getData(){
            return data;
        }
        private Node getNext(){
            return next;
        }
        private void setData(T newData){
            data = newData;
        }
        private void setNext(Node nextNode){
            next = nextNode;
        }

    }
}

File: Stack/Python/LinkedStack.py
class Node:
    def __init__(self,data,node):
        self.data = data
        self.next = node
    
class Stack:
    def __init__(self):
        self.size = 0
        self.top = None

    def pop(self):
            if not self.length():
                print("Empty Stack.")
                return
            temp = self.top.data
            self.top = self.top.next
            self.size -= 1
            return temp
    def push(self,data):
            self.top = Node(data,self.top)
            self.size += 1
    def clear(self):
        self.size = 0
        self.top = None

    def peek(self):
        if not self.length():
            print("Stack Empty peek.")
            return
        return self.top.data

    def length(self):
        return self.size
    def print_stack(self):
        if not self.length():
            print("Empty Stack.")
            return
        temp = self.top
        while temp:
            print(temp.data)
            temp = temp.next


if __name__ == "__main__":
    stack = Stack()
    stack.pop()
    stack.print_stack()
    stack.peek()
    for i in range(10):
        stack.push(i)
    print("Print Stack")
    stack.print_stack()
    print("Stack Length: "+str(stack.length()))
    print("Stack Peek: "+str(stack.peek()))
    print("Stack Pop Value: "+ str(stack.pop()))
    print("Stack Pop Value: "+ str(stack.pop()))
    print("Stack Length: "+str(stack.length()))
    print("Print Stack")
    stack.print_stack()
    print("Stack Clear")
    stack.clear()
    print("Stack Length: "+str(stack.length()))
    stack.peek()
    stack.pop()
    print("Print Stack")
    stack.print_stack()

